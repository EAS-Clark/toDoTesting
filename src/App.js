import { useState } from "react";
import { Todos } from "./components/todos";

export default function App() {
  const [text, setText] = useState();
  const [todos, setTodos] = useState([]);

  const handleAddButtonClick = () => {
    setTodos([...todos, { text, complete: false }]);
    setText("");
  }

  const toggleTodo = (index) => {
    setTodos(todos.map((todo, idx) => idx === index ? { ...todo, complete: !todo.complete } : todo));
  }

  const handleInputChange = (e) => {
    setText(e.target.value);
  }

  return (
    <div>
      <input type="text" value={text} onChange={handleInputChange} />
      <button onClick={handleAddButtonClick}>add</button>
      <Todos todos={todos} toggleTodo={toggleTodo} />
    </div>
  )
}

