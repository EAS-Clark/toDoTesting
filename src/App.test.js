import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import React from "react";

import App from './App';

test('renders add button', () => {
  render(<App />);
  const linkElement = screen.getByRole('button', { name: /add/i })
  expect(linkElement).toBeInTheDocument();
});


test('renders the same as the last Snapshot', () => {
  const tree = renderer
    .create(<App />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});