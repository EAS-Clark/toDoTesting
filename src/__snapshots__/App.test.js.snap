// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the same as the last Snapshot 1`] = `
<div>
  <input
    onChange={[Function]}
    type="text"
  />
  <button
    onClick={[Function]}
  >
    add
  </button>
  <div
    className="todos"
  />
</div>
`;
